SimpleOAuth/SimpleOAuth/ConfigurationModel/OAuthClientBaseConfiguration.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace SimpleOAuth
+{
+    /// <summary>
+    /// A base class that is used for all the configurations that are going to be used
+    /// </summary>
+    public abstract class OAuthClientBaseConfiguration
+    {
+        public OAuthClientBaseConfiguration(string clientId, string clientSecret)
+        {
+            ClientId = clientId;
+            ClientSecret = clientSecret;
+        }
+
+        public string ClientId { get; set; }
+        public string ClientSecret { get; set; }
```

Comment:
are there hurdles making the props immutable?
```suggestion
        public string ClientId { get;  }
        public string ClientSecret { get; }
```

Reply:
I can fix it as part of an other PR